tests/http_server_test.cc

```diff
@@ -986,23 +986,28 @@ TEST(http_server_test, client_request_timeout_on_delay_in_request_line_send_rais
     const std::string reqStr { "GET /ping HTTP/1.1\r\n" };
     TcpClient client;
     EXPECT_TRUE(client.connect(Pistache::Address("localhost", port))) << client.lastError();
+    bool send_failed = false;
     for (size_t i = 0; i < reqStr.size(); ++i)
     {
         if (!client.send(reqStr.substr(i, 1)))
         {
+            send_failed = true;
             break;
         }
         std::this_thread::sleep_for(std::chrono::milliseconds(300));
     }
 
-    EXPECT_EQ(client.lastErrno(), EPIPE) << "Errno: " << client.lastErrno();
+    if (send_failed)
+    { // Usually, send does fail; but on macOS occasionally it does not fail
+        EXPECT_EQ(client.lastErrno(), EPIPE) << "Errno: " << client.lastErrno();
```

Comment:
Is this a workaround or expected behaviour? Please expand this comment so that it is obvious when reading the source code